TOY-24 카메라 촬영 화면 UI 디자인 패턴 적용 및 추가 기능 구현

pby2017 opened the pull request:
코드 리뷰 부탁드립니다.

1. 카메라 그리드 및 캡처 크기 변경 기능 추가와 버튼의 배치를 수정했습니다.
2. 텍스트 및 아이콘 색변경 코드를 추가하고, 갤러리 버튼 원형으로 수정했습니다.
3. snackbar를 추가했습니다.

현재 CameraView를 ConstraintLayout로 감싸도록 수정했지만, 위 아래 비율을 조절하지 않아 특정 비율을 제공하지 않는 단말의 경우 CameraView와 컨트롤 버튼이 겹쳐서 의도하지 않은 뷰를 보여주고 있습니다.
티켓을 생성해서 해결하겠습니다.

pby2017 committed:
Merge branch 'TOY-25-camera-mvvm-kotlin' into TOY-24-camera-ui-feature

pby2017 committed:
TOY-24 카메라 기능 구현(그리드, 크기 변경) 및 기능 배치 수정

pby2017 committed:
TOY-24 Merge branch 'develop' into TOY-24-camera-ui-feature

# Conflicts:
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/BaseActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/Constants.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/capture/PreviewActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/DiaryDetailActivity.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/diary/DiaryEditActivity.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/edit/EditPhotoActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/edit/EditResultActivity.kt
#	app/src/main/res/layout/activity_camera.xml
#	app/src/main/res/values/strings.xml

pby2017 committed:
TOY-24 타이머 숫자 색 변경, 다이어리 아이콘 추가, 갤러리 모양 원형으로 변경

pby2017 committed:
TOY-24 snackbar 추가, 뒤로가기 버튼 시간차 종료 추가, loadImage 메서드 수정

pby2017 committed:
TOY-24 CameraView에 ConstraintLayout parent 적용 (위치 조정을 위해서 적용)

pby2017 committed:
TOY-24 갤러리 버튼 흑백 색 전환 및 사진 보여주기 문제 수정

hyejin830 commented on .idea/codeStyles/Project.xml:
```diff
@@ -0,0 +1,29 @@
+<component name="ProjectCodeStyleConfiguration">
```
이 파일은 삭제하는게 좋을 것 같습니다.

pby2017 replied:
네,
피드백 반영했습니다.

pby2017 committed:
TOY-24 그리드 포커스 제거

pby2017 committed:
TOY-24 Project.xml 파일 제거

KimSoungRyoul reviewed (approved):
제 Approve는 노카운트해주세요

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraViewModel.kt:
```diff
@@ -3,69 +3,111 @@ package kr.co.yogiyo.rookiephotoapp.camera
     }
 
     lateinit var captureNow: () -> Unit
+    lateinit var updateGalleryButton: CameraViewModel.() -> Unit
```
왜 CameraViewModel.() -> Unit 이렇게 작성한거에요?

pby2017 replied:
의도는 CameraActivity에서 CameraViewModel의 `isCaptureSizeFull()` 메서드를 `viewModel.` 키워드 없이 호출하려고 했습니다. 

그런데 제가 CameraActivity 코드에 `viewModel.isCaptureSizeFull()` 이렇게 직접 접근하도록 작성을 해놓아서 `CameraViewModel.() -> Unit` 은 필요가 없는 코드였네요.

`() -> Unit` 으로 수정하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraViewModel.kt:
```diff
@@ -3,69 +3,111 @@ package kr.co.yogiyo.rookiephotoapp.camera
     }
 
-    fun getNextCaptureDelay() = DELAY_DURATIONS[++delayDurationIndex % DELAY_DURATIONS.size]
+    fun synchronizeGalleryButton() = updateGalleryButton()
```
이 함수 필요한건가요?

pby2017 replied:
아뇨,
확인해보니 필요없습니다.

`CameraViewModel`에서
updateGalleryButton() 메서드를 CameraViewModel.() -> Unit 으로 정의하다보니 
updateGalleryButton() 메서드를 `CameraActivity`에서 호출하지 못해 

`CameraViewModel`에 synchronizeGalleryButton()를 새로 정의해서 CameraActivity에서 호출했었습니다.

`CameraViewModel.() -> Unit` 을 `() -> Unit` 으로 바꾸면서 synchronizeGalleryButton() 가 불필요한 코드가 되서 제거하겠습니다.

bonchangkoo commented on app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraViewModel.kt:
```diff
@@ -3,69 +3,111 @@ package kr.co.yogiyo.rookiephotoapp.camera
 import kr.co.yogiyo.rookiephotoapp.R
 
-class CameraViewModel(context: Context) : BaseObservable() {
+class CameraViewModel(private var context: Context) : BaseObservable() {
```
var은 웬만하면 쓰지 마세요~

pby2017 replied:
네,
var에 대해 이전에도 말씀해주셨는데 다시 한 번 숙지하겠습니다.

bonchangkoo reviewed (commented):
코멘트 확인 부탁드려요.

pby2017 committed:
TOY-24 피드백 반영

pby2017 committed:
TOY-24 충돌해결

Merge branch 'develop' into TOY-24-camera-ui-feature

# Conflicts:
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/BaseActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/Constants.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/MainActivity.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/camera/CameraActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/edit/EditPhotoActivity.kt
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryActivity.java
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/gallery/GalleryAdapter.java

pby2017 commented:
@bonchangkoo 님,

코멘트 확인 후 피드백 반영했습니다.
확인 부탁드립니다.

bonchangkoo reviewed (approved):
코드 확인했습니다.

pby2017 committed:
TOY-24 초기 카메라 비율 변경 (3:4), 편집 화면 배경색 제거

pby2017 committed:
TOY-24 충돌 해결

Merge branch 'develop' into TOY-24-camera-ui-feature

# Conflicts:
#	app/src/main/java/kr/co/yogiyo/rookiephotoapp/BaseActivity.kt
#	app/src/main/res/layout/dialog_fragment_backup_restore.xml
#	app/src/main/res/layout/dialog_fragment_sign.xml
#	app/src/main/res/layout/dialog_fragment_signout.xml
#	app/src/main/res/values/colors.xml
#	app/src/main/res/values/strings.xml
#	app/src/main/res/values/styles.xml
